src/MBC_UserImport_Source_Niche.php

```diff
@@ -254,209 +195,270 @@ public function setter($message)
-        throw new Exception(
-          'MBC_UserImport_Source_Niche->process() - No Drupal Id provided by Northstar.'
-          . ' Response: ' . var_export($northstarUser, true)
+      // Save mobile number to record loaded by email.
+      if (!empty($input['mobile'])) {
+        self::log(
+          'Updating user %s mobile phone from "%s" to "%s"',
+          $identity->id,
+          ($identity->mobile ?: "NULL"),
+          $input['mobile']
         );
+
+        $params = ['mobile' => $input['mobile']];
+        $identity = $northstar->updateUser($identity->id, $params);
       }
+    } elseif (!empty($identityByMobile) && empty($identityByEmail)) {
+      // ****** Existing user: only mobile record exists ******
+      $identity = &$identityByMobile;
+      self::log(
+        'User identified by mobile %s as %s',
+        $input['mobile'],
+        $identity->id
+      );
 
-      $this->addImportUserInfo($northstarUser);
-      $drupalUID = $northstarUser->drupal_id;
-      $passwordResetURL = $this->mbToolbox->getPasswordResetURL($drupalUID);
-      // #1, user_welcome, New/New
-      $payload['email_template'] = self::WELCOME_EMAIL_NEW_NEW;
-      $payload['tags'][0] = 'user-welcome-niche';
-      $payload['tags'][1] = self::WELCOME_EMAIL_NEW_NEW;
-      $payload['merge_vars']['PASSWORD_RESET_LINK'] = $passwordResetURL;
-    } else {
-      // Existing Drupal user. Set UID for campaign signup
-      $drupalUID = $existing['drupal-uid'];
-      // #2, current_user, Existing/New
-      $payload['email_template'] = self::WELCOME_EMAIL_EXISTING_NEW;
-      $payload['tags'][0] = 'current-user-welcome-niche';
-      $payload['tags'][1] = self::WELCOME_EMAIL_EXISTING_NEW;
-    }
+      // Save email to record loaded by mobile.
+      self::log(
+        'Updating user %s email from "%s" to "%s"',
+        $identity->id,
+        ($identity->email ?: "NULL"),
+        $input['email']
+      );
+      $params = ['email' => $input['email']];
+      $identity = $northstar->updateUser($identity->id, $params);
+    } elseif ($identityByEmail->id !== $identityByMobile->id) {
+      // ****** Existing users: loaded both by mobile and phone ******
+      // We presume that user account with mobile number generally have
+      // email address as well. For this reason we decided to use
+      // identity loaded by mobile rather than by email.
+      $identity = &$identityByMobile;
+
+      self::log(
+        'User identified by email %s as %s and by mobile %s as %s'
+          . ' Selecting mobile identity',
+        $input['mobile'],
+        $identityByMobile->id,
+        $input['email'],
+        $identityByEmail->id
+      );
 
-    // Campaign signup
-    $campaignNID = self::PHOENIX_SIGNUP;
-    $campaignSignup = $this->mbcUserImportToolbox->campaignSignup(
-      $campaignNID,
-      $drupalUID,
-      'niche',
-      false
-    );
+    } elseif ($identityByEmail->id === $identityByMobile->id) {
```

Comment:
And if relying on the upsert functionality on that endpoint, you can then just use the HTTP status code to determine whether the account was newly created by this import: 200 for previously existed, 201 for new.